code/modules/mob/living/say.dm

```diff
@@ -245,9 +245,9 @@ GLOBAL_LIST_INIT(department_radio_keys, list(
 
 
 /mob/living/proc/get_key(message)
-	var/key = message[1]
-	if(key in GLOB.department_radio_prefixes)
-		return lowertext(message[1 + length(key)])
+	var/prefix = message[1]
+	if(length(message) >= 2 && (prefix in GLOB.department_radio_prefixes))
+		return lowertext(message[2])
```

Comment:
Couold this possibly break ;

Reply:
Actually could maybe break :b and two or more characters i think

Reply:
I don't see how? It's doing exactly what it did before?

`message[1 + length(key)]` = `meessage[2]` when key can only ever be a length of 1 since it is a char index too. Think maints!

Reply:
There are no 2 letter department keys we checked when I did the same fix to `get_message_language()`

Even if there were the old code would not work with them either.

Reply:
> I don't see how? It's doing exactly what it did before?
> 
> `message[1 + length(key)]` = `meessage[2]` when key can only ever be a length of 1 since it is a char index too. Think maints!

; is a one letter key did you test this

Reply:
Yes I did, now if you attempt to say just `;` you transmit `.` over the radio. 

Plus `get_key()` does nothing for `;` anyway besides causing a runtime if you send just one character.